lib/std/Target.zig

```diff
@@ -1997,14 +1968,38 @@ pub const Cpu = struct {
             .fs, .gs, .ss => (arch == .x86_64 or arch == .x86) and (context == null or context == .pointer),
             .flash, .flash1, .flash2, .flash3, .flash4, .flash5 => arch == .avr, // TODO this should also check how many flash banks the cpu has
             .cog, .hub => arch == .propeller,
-            .lut => arch == .propeller and std.Target.propeller.featureSetHas(cpu.features, .p2),
+            .lut => arch == .propeller and cpu.has(.propeller, .p2),
 
             .global, .local, .shared => is_gpu,
             .constant => is_gpu and (context == null or context == .constant),
             .param => is_nvptx,
             .input, .output, .uniform, .push_constant, .storage_buffer => is_spirv,
         };
     }
+
+    /// Returns true if `feature` is enabled.
+    pub fn has(cpu: Cpu, comptime family: Arch.Family, feature: @field(Target, @tagName(family)).Feature) bool {
+        if (family != cpu.arch.family()) return false;
```

Comment:
I'm a bit torn between the current behavior here and `assert(family == cpu.arch.family())`. I like that the current behavior lets you check the family and feature(s) at the same time, but I can also see an argument that asserting might help catch bugs. :thinking:

Reply:
A quick grep shows a bunch of `if (is_xyz_cpu) std.Target.xyz.featureSetHas()` instances and honestly, I can't imagine a scenario where asserting could be helpful to catch a bug.